src/main/java/org/opentripplanner/routing/edgetype/TripPattern.java

```diff
@@ -498,33 +497,15 @@ public static void generateUniqueNames (Collection<TripPattern> tableTripPattern
     }
 
     /**
-     * Repetitive logic pulled out of makePatternVerticesAndEdges().
-     * No longer works because we don't have access to the DAO here.
-     * But moving the makePatternVerticesAndEdges into TripPattern seems cleaner (certainly looks cleaner).
-     */
-    private <T> T getStopOrParent(Map<Stop, T> map, Stop stop, Graph graph) {
-        T vertex = map.get(stop);
-        if (vertex == null) {
-            Stop parent = null; //_dao.getStopForId(new AgencyAndId(stop.getId().getAgencyId(), stop.getParentStation()));
-            vertex = map.get(parent);
-            /* FIXME: this is adding an annotation for a specific problem, but all we know is that the stop vertex does not exist. */
-            if (vertex == null) {
-                //LOG.warn(graph.addBuilderAnnotation(new StopAtEntrance(stop, false)));
-            } else {
-                //LOG.warn(graph.addBuilderAnnotation(new StopAtEntrance(stop, true)));
-            }
-        }
-        return vertex;
-    }
-
-    /**
      * Create the PatternStop vertices and PatternBoard/Hop/Dwell/Alight edges corresponding to a
      * StopPattern/TripPattern. StopTimes are passed in instead of Stops only because they are
      * needed for shape distances (actually, stop sequence numbers?).
      *
-     * TODO move GtfsStopContext into Graph.
+     * @param graph graph to create vertices and edges in
+     * @param transitStops map of transit stops given the stop; it is assumed all stops of this trip
+     *        pattern are included in this map and refer to TransitStops
      */
-    public void makePatternVerticesAndEdges(Graph graph, GtfsStopContext context) {
+    public void makePatternVerticesAndEdges(Graph graph, Map<Stop, ? extends TransitStationStop> transitStops) {
```

Comment:
I know this is not your doing, but it's strange that we allow Stations here and not just Stops. I think it's a mistake in GTFS if a trip passes through a Station rather than a Stop.